### PR TITLE
feat: add minectl cli spec

### LINKED ISSUE #1434: [minectl] Add minctl support
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [X] I would like to work on this.

### What CLI tool does this relate to?

minectl

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

New Spec for the [minectl cli ](https://github.com/dirien/minectl)

### PR DESCRIPTION
**Pull request by dirien**
Signed-off-by: Engin Diri <engin.diri@mail.schwarz>

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

Adds support for `minectl`, closes #1434

### CODE REVIEW

File: src/minectl.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+    {
+      name: "--log-encoding",
+      description: "Set the log encoding: console|json (default: console)",
+      isPersistent: true,
+      args: { name: "log-encoding", default: "console" },
+    },
+    {
+      name: "--verbose",
+      description:
+        "Enable verbose logging: debug|info|warn|error|dpanic|panic|fatal",
```
Comment (fedeci) [inline src/minectl.ts:236]: This can be listed using the `suggestions` prop under the child argument of this option.

File: src/minectl.ts [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+      name: "--headless",
+      description:
+        "Set this value to if mincetl is called by a CI system. Enables logging and disables human-readable output rendering (default: false)",
+      isPersistent: true,
+    },
+    {
+      name: "--log-encoding",
+      description: "Set the log encoding: console|json (default: console)",
+      isPersistent: true,
+      args: { name: "log-encoding", default: "console" },
```
Comment (fedeci) [inline src/minectl.ts:231]: This too can use the `suggestions` prop

File: src/minectl.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+        {
+          name: "version",
+          description: "Display the clients version information",
+        },
+        {
+          name: "wizard",
+          description:
+            "Calls the minectl wizard to create interactively a minectl 🗺 config",
+        },
+      ],
```
Comment (fedeci) [inline src/minectl.ts:217]: This template is a nice sugar that does the same thing requiring not maintenance
```suggestion
      args: {
        template: "help",
      },
```

File: src/minectl.ts [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+    },
+    {
+      name: "rcon",
+      description: "RCON client to your Minecraft server",
+      options: [
+        {
+          name: ["--filename", "-f"],
+          description: "Contains the configuration for minectl",
+          args: { name: "filename" },
```
Comment (fedeci) [inline src/minectl.ts:133]:
```suggestion
          args: { name: "filename", template: "filepaths" },
```

File: src/minectl.ts [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+    {
+      name: "wizard",
+      description:
+        "Calls the minectl wizard to create interactively a minectl 🗺 config",
+      options: [
+        {
+          name: ["--output", "-o"],
+          description:
+            "Output folder for the configuration file for minectl 🗺 (default: /Users/dirien/.minectl)",
+          args: { name: "output" },
```
Comment (fedeci) [inline src/minectl.ts:168]:
```suggestion
          args: { name: "output", template: "folders", suggestCurrentToken: true },
```

File: src/minectl.ts [hunk 6]
```diff
@@ -0,0 +1,243 @@
...
+      ],
+    },
+    {
+      name: "update",
+      description: "Update an Minecraft Server",
+      options: [
+        {
+          name: ["--filename", "-f"],
+          description: "Contains the configuration for minectl",
+          args: { name: "filename" },
```
Comment (fedeci) [inline src/minectl.ts:149]:
```suggestion
          args: { name: "filename", template: "filepaths" },
```

File: src/minectl.ts [hunk 7]
```diff
@@ -0,0 +1,243 @@
...
+        },
+        {
+          name: "--id",
+          description: "Contains the server id",
+          args: { name: "id" },
+        },
+        {
+          name: ["--plugin", "-p"],
+          description: "Local plugin file location",
+          args: { name: "plugin" },
```
Comment (fedeci) [inline src/minectl.ts:122]:
```suggestion
          args: { name: "plugin", template: "filepaths" },
```

File: src/minectl.ts [hunk 8]
```diff
@@ -0,0 +1,243 @@
...
+      options: [
+        {
+          name: ["--destination", "-d"],
+          description: "Plugin destination location",
+          args: { name: "destination" },
+        },
+        {
+          name: ["--filename", "-f"],
+          description: "Contains the configuration for minectl",
+          args: { name: "filename" },
```
Comment (fedeci) [inline src/minectl.ts:112]:
```suggestion
          args: { name: "filename", template: "filepaths" },
```

File: src/minectl.ts [hunk 9]
```diff
@@ -0,0 +1,243 @@
...
+      ],
+    },
+    {
+      name: "plugins",
+      description: "Manage your plugins for a specific server",
+      options: [
+        {
+          name: ["--destination", "-d"],
+          description: "Plugin destination location",
+          args: { name: "destination" },
```
Comment (fedeci) [inline src/minectl.ts:107]:
```suggestion
          args: { name: "destination", template: "folders" },
```

File: src/minectl.ts [hunk 10]
```diff
@@ -0,0 +1,243 @@
...
+    },
+    {
+      name: "list",
+      description: "List all Minecraft Server",
+      options: [
+        {
+          name: ["--provider", "-p"],
+          description:
+            "The cloud provider - civo|scaleway|do|hetzner|linode|ovh|equinix|gce|vultr|azure|oci|ionos|aws|vexxhost|multipass|exoscale",
+          args: { name: "provider" },
```
Comment (fedeci) [inline src/minectl.ts:90]: This can be improved using a `suggestion` property too

File: src/minectl.ts [hunk 11]
```diff
@@ -0,0 +1,243 @@
...
+      ],
+    },
+    {
+      name: "create",
+      description: "Create an Minecraft Server",
+      options: [
+        {
+          name: ["--filename", "-f"],
+          description: "Contains the configuration for minectl",
+          args: { name: "filename" },
```
Comment (fedeci) [inline src/minectl.ts:58]:
```suggestion
          args: { name: "filename", template: "filepaths" },
```

File: src/minectl.ts [hunk 12]
```diff
@@ -0,0 +1,243 @@
...
+      ],
+    },
+    {
+      name: "delete",
+      description: "Delete an Minecraft Server",
+      options: [
+        {
+          name: ["--filename", "-f"],
+          description: "That contains the configuration for minectl",
+          args: { name: "filename" },
```
Comment (fedeci) [inline src/minectl.ts:73]:
```suggestion
          args: { name: "filename", template: "filepaths" },
```

**Review by fedeci — CHANGES_REQUESTED** [review]
Thanks a lot for this specs @dirien! I left some comments to solve before merging this. Likely prettier will have to complain if you commit from github, you can fix those problems running `npm run lint:fix` and committing the changes

File: src/minectl.ts (hunk 1 above)
Comment (dirien) [reply src/minectl.ts:236]: done

File: src/minectl.ts (hunk 2 above)
Comment (dirien) [reply src/minectl.ts:231]: done

**Comment by dirien** [comment]
Hi @fedeci,

thanks for the feedback! I changed all parts you highlighted!

**Review by fedeci — APPROVED** [review]
Thanks!

**Pull request merged**